php/class-plugin.php

```diff
@@ -29,21 +29,46 @@ class Plugin extends Plugin_Base {
 	public $version;
 
 	/**
+	 * Is old version of WordPress.
+	 *
+	 * @var boolean
+	 */
+	public $compat;
+
+	/**
+	 * Migration handler.
+	 *
+	 * @var Migrate
+	 */
+	public $migrate;
+
+	/**
 	 * Plugin constructor.
 	 */
 	public function __construct() {
-
 		// Parse plugin version.
 		if ( preg_match( '/Version:\s*(\S+)/', file_get_contents( __DIR__ . '/../customize-snapshots.php' ), $matches ) ) {
 			$this->version = $matches[1];
 		}
-
+		$this->compat = is_back_compat();
 		load_plugin_textdomain( 'customize-snapshots' );
 
 		parent::__construct();
 	}
 
 	/**
+	 * Init migration.
+	 *
+	 * @action init
+	 */
+	public function init_migration() {
+		$this->migrate = new Migrate();
```

Comment:
You could pass `$this` to `Migrate` and then have access to `$compat` instead of recreating it.